clusterloader2/pkg/measurement/common/metrics_for_e2e.go

```diff
@@ -0,0 +1,209 @@
+	if err != nil {
+		return summaries, err
+	}
+
+	grabMetricsFromKubelets := strings.ToLower(provider) != "kubemark"
+	switch action {
+	case "reset":
+		if includeClusterAutoscalerMetrics {
+			grabber, err := metrics.NewMetricsGrabber(config.ClientSet, nil, grabMetricsFromKubelets, false, false, false, true)
+			if err != nil {
+				glog.Infof("Failed to create MetricsGrabber (skipping ClusterAutoscaler metrics gathering before test): %v", err)
+			} else {
+				m.clusterAutoscalerMetricsBeforeTest, err = grabber.Grab()
+				if err != nil {
+					glog.Infof("MetricsGrabber failed to grab CA metrics before test (skipping metrics gathering): %v", err)
+				} else {
+					glog.Infof("Gathered ClusterAutoscaler metrics before test")
+				}
+			}
+
+		}
+	case "gather":
+		glog.Infof("Gathering metrics")
+		// Grab apiserver, scheduler, controller-manager metrics and (optionally) nodes' kubelet metrics.
+
+		grabber, err := metrics.NewMetricsGrabber(config.ClientSet, nil, grabMetricsFromKubelets, true, true, true, includeClusterAutoscalerMetrics)
+		if err != nil {
+			glog.Infof("Failed to create MetricsGrabber (skipping metrics gathering): %v", err)
+		} else {
+			received, err := grabber.Grab()
+			if err != nil {
+				glog.Infof("MetricsGrabber failed to grab some of the metrics: %v", err)
+			}
+			(*MetricsForE2E)(&received).computeClusterAutoscalerMetricsDelta(m.clusterAutoscalerMetricsBeforeTest)
+			summaries = append(summaries, (*MetricsForE2E)(&received))
+		}
+	default:
+		return summaries, fmt.Errorf("unknown action %v", action)
+	}
+	return summaries, nil
+}
+
+type MetricsForE2E metrics.MetricsCollection
+
+func (m *MetricsForE2E) filterMetrics() {
+	interestingApiServerMetrics := make(metrics.ApiServerMetrics)
+	for _, metric := range InterestingApiServerMetrics {
+		interestingApiServerMetrics[metric] = (*m).ApiServerMetrics[metric]
+	}
+	interestingControllerManagerMetrics := make(metrics.ControllerManagerMetrics)
+	for _, metric := range InterestingControllerManagerMetrics {
+		interestingControllerManagerMetrics[metric] = (*m).ControllerManagerMetrics[metric]
+	}
+	interestingClusterAutoscalerMetrics := make(metrics.ClusterAutoscalerMetrics)
+	for _, metric := range InterestingClusterAutoscalerMetrics {
+		interestingClusterAutoscalerMetrics[metric] = (*m).ClusterAutoscalerMetrics[metric]
+	}
+	interestingKubeletMetrics := make(map[string]metrics.KubeletMetrics)
+	for kubelet, grabbed := range (*m).KubeletMetrics {
+		interestingKubeletMetrics[kubelet] = make(metrics.KubeletMetrics)
+		for _, metric := range InterestingKubeletMetrics {
+			interestingKubeletMetrics[kubelet][metric] = grabbed[metric]
+		}
+	}
+	(*m).ApiServerMetrics = interestingApiServerMetrics
+	(*m).ControllerManagerMetrics = interestingControllerManagerMetrics
+	(*m).KubeletMetrics = interestingKubeletMetrics
+}
+
+func (m *MetricsForE2E) computeClusterAutoscalerMetricsDelta(before metrics.MetricsCollection) {
```

Comment:
This can be removed as I think we're not running cluster autoscaler in our tests anyway (or collecting/plotting any of it's metrics).

Reply:
Not really.. There's still 'caFunctionMetric' and something else around that. Please remove.

Reply:
Please fix this.

Reply:
You still haven't removed all CA related code. Please remove. There are things like `caFunctionMetric`, etc in the code.

Reply:
I removed the constants. I think that they were the last things related to the CA.